### PR TITLE
Moves intern and label modules into bevy_ecs

### PR DESCRIPTION
**Pull request by mnmaita**
# Objective

- Attempts to solve two items from https://github.com/bevyengine/bevy/issues/11478.

## Solution

- Moved `intern` module from `bevy_utils` into `bevy_ecs` crate and updated all relevant imports.
- Moved `label` module from `bevy_utils` into `bevy_ecs` crate and updated all relevant imports.

---

## Migration Guide

- Replace `bevy_utils::define_label` imports with `bevy_ecs::define_label` imports.
- Replace `bevy_utils::label::DynEq` imports with `bevy_ecs::label::DynEq` imports.
- Replace `bevy_utils::label::DynHash` imports with `bevy_ecs::label::DynHash` imports.
- Replace `bevy_utils::intern::Interned` imports with `bevy_ecs::intern::Interned` imports.
- Replace `bevy_utils::intern::Internable` imports with `bevy_ecs::intern::Internable` imports.
- Replace `bevy_utils::intern::Interner` imports with `bevy_ecs::intern::Interner` imports.

### CODE REVIEW

**Review by james7132 — COMMENTED** [review]
Mind fixing CI? The lints seem like they should be fixable by using `cast::<()>` instead of `as`.

**Comment by mnmaita** [comment]
> Mind fixing CI? The lints seem like they should be fixable by using `cast::<()>` instead of `as`.

Done, rebased and updated the migration guide. Feel free to take a second look at it now @james7132. Thanks!

File: crates/bevy_ecs/src/label.rs [hunk 1]
```diff
@@ -125,7 +125,7 @@ macro_rules! define_label {
             /// Feeds this value into the given [`Hasher`].
             fn dyn_hash(&self, state: &mut dyn ::std::hash::Hasher);
 
-            /// Returns an [`Interned`](bevy_utils::intern::Interned) value corresponding to `self`.
+            /// Returns an [`Interned`](bevy_ecs::intern::Interned) value corresponding to `self`.
```
Comment (mnmaita) [inline crates/bevy_ecs/src/label.rs:128]: Changing this to `[`Interned`](Interned)` fixes one of the issues in the docs CI job but there seems to be another issue that I couldn't figure out yet. I'd appreciate some guidance with it.

Comment (james7132) [reply crates/bevy_ecs/src/label.rs:128]:
```suggestion
            /// Returns an [`Interned`](crate::intern::Interned) value corresponding to `self`.
```

`crate` is self-referential when inside the target crate.

**Review by james7132 — APPROVED** [review]

File: crates/bevy_ecs/src/label.rs (hunk 1 above)
Comment (mnmaita) [reply crates/bevy_ecs/src/label.rs]: Thanks for applying that change. I'm 99% positive that I tried that and it error-ed out locally but I might be wrong. Let's see if that fixes the CI issue.

**Comment by mnmaita** [comment]
@james7132 I did a change that fixed the issue with `Interned` in the docs, but now there's a different one... Could you please help me figure out why `Schedule` is giving errors during doc generation? I don't really know what's the cause of this issue. Thanks!

File: crates/bevy_ecs/src/schedule/set.rs [hunk 2]
```diff
@@ -3,14 +3,16 @@ use std::fmt::Debug;
...
+use crate::{
+    define_label,
+    intern::Interned,
+    system::{
+        ExclusiveFunctionSystem, ExclusiveSystemParamFunction, FunctionSystem,
+        IsExclusiveFunctionSystem, IsFunctionSystem, SystemParamFunction,
+    },
 };
 
 define_label!(
```
Comment (james7132) [inline crates/bevy_ecs/src/schedule/set.rs:18]: I think you just need to make this ``[`Schedule`](crate::schedule::Schedule)``.

Comment (mnmaita) [reply crates/bevy_ecs/src/schedule/set.rs:18]: This should be fixed in the latest change @james7132

**Review by james7132 — APPROVED** [review]

**Pull request merged**